pkg/broker/cel.go

```diff
@@ -0,0 +1,137 @@
+package broker
+
+import (
+	"bytes"
+	"encoding/json"
+
+	"github.com/cloudevents/sdk-go/pkg/cloudevents"
+	"github.com/golang/protobuf/jsonpb"
+	structpb "github.com/golang/protobuf/ptypes/struct"
+	"github.com/google/cel-go/cel"
+	"github.com/google/cel-go/checker/decls"
+	"github.com/google/cel-go/common/types"
+	eventingv1alpha1 "github.com/knative/eventing/pkg/apis/eventing/v1alpha1"
+	celprotos "github.com/knative/eventing/pkg/broker/dev_knative"
+	"go.uber.org/zap"
+)
+
+const (
+	// CELVarKeyContext is the CEL variable key used for the CloudEvent event
+	// context fields defined in the spec.
+	CELVarKeyContext = "ce"
+	// CELVarKeyExtensions is the CEL variable key used for the CloudEvent event
+	// context extensions.
+	CELVarKeyExtensions = "ext"
+	// CELVarKeyData is the CEL variable key used for the CloudEvent event data.
+	CELVarKeyData = "data"
+	//TODO add a key that contains both the extensions and the baseline context
```

Comment:
This seems very important to me. 

If we don't do this, then what happens when I have running filters while updating to a new Knative eventing release, which happens to use the next version of the CE spec?

I think that would cause all my filters to stop working. In fact, to do upgrade safely, I would need to modify every filter, duplicating the existing logic that uses an extension attribute, changing it to use the standard attribute, then OR those two together.

Reply:
This is a great point. The immediate example is subject. As written, a `subject` attribute would be available as `ext.subject`, until this code decides to elevate it to an official attribute at which point it becomes `ce.subject`.

I'd call this a blocker to merging. See the similar discussion at https://github.com/knative/eventing/pull/1047#discussion_r274706752.

Reply:
This is fixed by 11710b5829813bb9707426bebfe216c2c5134cad. Extensions, when parsed, are available under the `ce` key.